src/stcal/resample/resample.py

```diff
@@ -0,0 +1,1660 @@
+
+            output_variance = (
+                self._var_poisson_wsum / (self._var_poisson_weight *
+                                          self._var_poisson_weight)
+            ).astype(
+                dtype=self.output_array_types["var_poisson"]
+            )
+            output_model["var_poisson"] = output_variance
+
+            output_variance = (
+                self._var_flat_wsum / (self._var_flat_weight *
+                                       self._var_flat_weight)
+            ).astype(
+                dtype=self.output_array_types["var_flat"]
+            )
+            output_model["var_flat"] = output_variance
+
+        if free_memory:
+            self._finalized = True
+            del (
+                self._var_rnoise_wsum,
+                self._var_poisson_wsum,
+                self._var_flat_wsum,
+                self._var_rnoise_weight,
+                self._var_poisson_weight,
+                self._var_flat_weight,
+            )
+
+    def _resample_one_variance_array(self, name, model, iscale,
+                                     weight_map, pixmap,
+                                     xmin=None, xmax=None, ymin=None,
+                                     ymax=None):
+        """Resample one variance image from an input model.
+
+        The error image is passed to drizzle instead of the variance in order
+        to better match kernel overlap and user weights to the data during the
+        pixel averaging process. The drizzled error image is squared before
+        returning.
+
+        """
+        variance = model.get(name)
+        if variance is None or variance.size == 0:
+            log.debug(
+                f"No data for '{name}' for model "
+                f"{repr(model['filename'])}. Skipping ..."
+            )
+            return
+
+        elif variance.shape != model["data"].shape:
+            log.warning(
+                f"Data shape mismatch for '{name}' for model "
+                f"{repr(model['filename'])}. Skipping ..."
+            )
+            return
+
+        output_shape = self.output_array_shape
+
+        # Resample the error array. Fill "unpopulated" pixels with NaNs.
+        driz = Drizzle(
+            out_shape=output_shape,
+            kernel=self.kernel,
+            fillval=np.nan,
+            disable_ctx=True
+        )
+
+        # Call 'drizzle' to perform image combination
+        log.info(f"Drizzling {variance.shape} --> {output_shape}")
+
+        driz.add_image(
+            data=np.sqrt(variance),
+            exptime=model["exposure_time"],
+            pixmap=pixmap,
+            scale=iscale,
+            weight_map=weight_map,
+            wht_scale=1.0,
+            pixfrac=self.pixfrac,
+            in_units="cps",
+            xmin=xmin,
+            xmax=xmax,
+            ymin=ymin,
+            ymax=ymax,
+        )
+
+        return driz.out_img ** 2
+
+    def build_driz_weight(self, model, weight_type=None, good_bits=None):
```

Comment:
`build_driz_weight` is currently a function provided by [resample_utils](https://github.com/spacetelescope/jwst/blob/f1cb0a61e9ee8989bc770157f297f64b1b7dbc68/jwst/resample/resample_utils.py#L194) (in jwst) and used by outlier detection.
It is used even if [no resampling is performed](https://github.com/spacetelescope/jwst/blob/f1cb0a61e9ee8989bc770157f297f64b1b7dbc68/jwst/outlier_detection/utils.py#L124).

By making it a method on the `ResampleData` class we'd have to either keep the duplicated code or construct a `ResampleData` instance (which would allocate an unused output model etc).

Does this need to be a method on `ResampleData`?

Reply:
moved it `build_driz_weight` to `stcal.resample.utils`. `ResampleData` does not exist, I think...